### PR TITLE
Fix issue with no decompressed data in ORC reader

### PR DESCRIPTION
**Pull request by vuule**
## Description
Currently, the ORC reader assumes that if data in the stripes of the current level are not empty, the decompressed data will also not be empty.
However, there is a corner case where the stripe is empty, but data blocks are still compressed so they contain the compression header. In this case, decompressed data is empty even with non-empty compressed blocks. 

This PR removes the assertion in the reader to allow for this corner case. Also adds a short-circuit to `decompress_stripe_data` to return early if the decompressed data is empty.

Issue https://github.com/rapidsai/cudf/issues/13608

## Checklist
- [x] Verify to fix the issue.
- [X] Check with compute-sanitizer.
- [X] Test with spark-rapids integration tests.

### CODE REVIEW

**Comment by vuule** [comment]
Need to add a test, but still not sure what corner case triggers the issue.

**Comment by ttnghia** [comment]
Confirm that this fixes the issue.

File: cpp/src/io/orc/reader_impl.cu [hunk 1]
```diff
@@ -317,9 +317,12 @@ rmm::device_buffer reader::impl::decompress_stripe_data(
     num_uncompressed_blocks += compinfo[i].num_uncompressed_blocks;
     total_decomp_size += compinfo[i].max_uncompressed_size;
   }
-  CUDF_EXPECTS(total_decomp_size > 0, "No decompressible data found");
+  CUDF_EXPECTS(
+    not((num_compressed_blocks + num_compressed_blocks > 0) and (total_decomp_size == 0)),
```
Comment (mythrocks) [inline cpp/src/io/orc/reader_impl.cu:321]: The cleverness of this bit is eluding me:
```c++
(num_compressed_blocks + num_compressed_blocks > 0) and ...
```
Surely `num_compressed_blocks > 0` suffices, no?

Comment (ttnghia) [inline cpp/src/io/orc/reader_impl.cu:321]: Should be this:
```suggestion
    not((num_compressed_blocks + num_uncompressed_blocks > 0) and (total_decomp_size == 0)),
```

Comment (ttnghia) [reply cpp/src/io/orc/reader_impl.cu:321]: Good catch. I also didn't realize that :D

Comment (mythrocks) [reply cpp/src/io/orc/reader_impl.cu:321]: Ok, that makes sense now.

Comment (vuule) [reply cpp/src/io/orc/reader_impl.cu]: the names are too similar at first glance :)
Sorry for the unintentional puzzle :D

**Review by ttnghia — APPROVED** [review]

**Review by bdice — APPROVED** [review]

**Review by PointKernel — APPROVED** [review]
